packages/redis/src/lib/RedisProvider.ts

```diff
@@ -0,0 +1,704 @@
+import {
+  CommonIdentifiers,
+  deleteProperty,
+  getProperty,
+  hasProperty,
+  isEveryByHookPayload,
+  isEveryByValuePayload,
+  isFilterByHookPayload,
+  isFilterByValuePayload,
+  isFindByHookPayload,
+  isFindByValuePayload,
+  isMapByHookPayload,
+  isMapByPathPayload,
+  isPartitionByHookPayload,
+  isPartitionByValuePayload,
+  isPayloadWithData,
+  isRemoveByHookPayload,
+  isRemoveByValuePayload,
+  isSomeByHookPayload,
+  isSomeByValuePayload,
+  JoshProvider,
+  MathOperator,
+  Method,
+  Payloads,
+  PROPERTY_NOT_FOUND,
+  setProperty
+} from '@joshdb/core';
+import { Serialize } from '@joshdb/serialize';
+import { isNullOrUndefined, isNumber, isPrimitive } from '@sapphire/utilities';
+import { createClient, RedisClientOptions, RedisClientType } from 'redis';
+import { v4 } from 'uuid';
```

Comment:
Does Redis have a feature for unique ids?

Reply:
Nope, it's pretty basic with set and get being primary ways of handling data